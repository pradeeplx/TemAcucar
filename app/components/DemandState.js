import React from 'react'
import { View, StyleSheet } from 'react-native'
import { fontFactor } from "../helpers"
import Colors from "../Colors"

export default DemandState = ({ state, style }) => (
  <View style={[{
    borderRadius: 4,
    borderColor: Colors.white,
    borderWidth: StyleSheet.hairlineWidth,
    paddingVertical: 2,
    paddingHorizontal: 6,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
  }, style]}>
    <Icon name={(state === 'completed' ? 'ios-checkmark-circle-outline' : (state === 'canceled' ? 'ios-close-circle-outline' : (state === 'flagged' ? 'ios-warning-outline' : 'ios-time-outline')))} style={{
      color: Colors.white,
      fontSize: 14 * fontFactor(),
      marginRight: 6,
    }} />
    <Sentence style={{
      fontSize: 8 * fontFactor(),
      color: Colors.white,
    }}>
      {(state === 'completed' ? 'Bem-sucedido' : (state === 'canceled' ? 'Cancelado' : (state === 'notifying' ? 'Perguntando pros vizinhos' : (state === 'flagged' ? 'Denunciado como impr√≥prio' : 'Em andamento'))))}
    </Sentence>
  </View>
)
