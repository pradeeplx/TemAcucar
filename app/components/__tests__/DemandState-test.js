import { View } from 'react-native'
import React from 'react'
import DemandState from '../DemandState'
import renderer from 'react-test-renderer'

for(const state of ['completed', 'canceled', 'flagged', 'foo']) {
  it(`renders correctly with state ${state}`, () => {
    const tree = renderer.create(
      <DemandState state={state} top={{top: 0}} />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
}
