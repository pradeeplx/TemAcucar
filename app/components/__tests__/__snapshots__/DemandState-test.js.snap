exports[`test renders correctly with state canceled 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Ionicons",
          "fontSize": 27,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 21,
          "marginRight": 6,
        },
      ]
    }>
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        },
      ]
    }>
    Cancelado
  </Text>
</View>
`;

exports[`test renders correctly with state completed 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Ionicons",
          "fontSize": 27,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 21,
          "marginRight": 6,
        },
      ]
    }>
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        },
      ]
    }>
    Bem-sucedido
  </Text>
</View>
`;

exports[`test renders correctly with state flagged 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Ionicons",
          "fontSize": 27,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 21,
          "marginRight": 6,
        },
      ]
    }>
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        },
      ]
    }>
    Denunciado como impróprio
  </Text>
</View>
`;

exports[`test renders correctly with state foo 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 4,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Ionicons",
          "fontSize": 27,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 21,
          "marginRight": 6,
        },
      ]
    }>
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        },
      ]
    }>
    Em andamento
  </Text>
</View>
`;
